server/app/services/question/QuestionOption.java

```diff
@@ -42,21 +42,61 @@ public abstract class QuestionOption {
   public static QuestionOption jsonCreator(
       @JsonProperty("id") long id,
       @JsonProperty(value = "displayOrder", defaultValue = "-1L") long displayOrder,
+      @JsonProperty("adminName") String adminName,
       @JsonProperty("localizedOptionText") LocalizedStrings localizedOptionText,
       @JsonProperty("optionText") ImmutableMap<Locale, String> legacyOptionText) {
     if (displayOrder == -1) {
       displayOrder = id;
     }
     if (localizedOptionText != null) {
-      return QuestionOption.create(id, displayOrder, localizedOptionText);
+      return QuestionOption.create(id, displayOrder, adminName, localizedOptionText);
     }
-    return QuestionOption.create(id, displayOrder, LocalizedStrings.create(legacyOptionText));
+    return QuestionOption.create(
+        id, displayOrder, adminName, LocalizedStrings.create(legacyOptionText));
   }
 
-  /** Create a QuestionOption. */
+  /**
+   * Create a {@link QuestionOption}.
+   *
+   * @param id the option id
+   * @param displayOrder the option display
+   * @param adminName the option's immutable admin name, exposed via the API
+   * @param optionText the option's user-facing text
+   * @return the {@link QuestionOption}
+   */
+  public static QuestionOption create(
+      long id, long displayOrder, String adminName, LocalizedStrings optionText) {
+    return QuestionOption.builder()
+        .setId(id)
+        .setAdminName(adminName)
+        .setOptionText(optionText)
+        .setDisplayOrder(OptionalLong.of(displayOrder))
+        .build();
+  }
+
+  /**
+   * Create a {@link QuestionOption}.
+   *
+   * @param id the option id
+   * @param adminName the option's immutable admin name, exposed via the API
+   * @param optionText the option's user-facing text
+   * @return the {@link QuestionOption}
+   */
+  public static QuestionOption create(long id, String adminName, LocalizedStrings optionText) {
+    return QuestionOption.builder()
+        .setId(id)
+        .setAdminName(adminName)
+        .setOptionText(optionText)
+        .setDisplayOrder(OptionalLong.empty())
+        .build();
+  }
+
+  /**
+   * Create a QuestionOption.
+   *
+   * <p>TODO(#4862): Deprecate this method.
+   */
   public static QuestionOption create(long id, long displayOrder, LocalizedStrings optionText) {
```

Comment:
I left these older `create()` methods in place so that I don't have to migrate everything at once.